tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -4633,6 +4661,39 @@ uint32_t SPIRVEmitter::processIntrinsicClamp(const CallExpr *callExpr) {
                                   {argXId, argMinId, argMaxId});
 }
 
+uint32_t SPIRVEmitter::processIntrinsicMemoryBarrier(const CallExpr *callExpr,
+                                                     bool isDevice,
+                                                     bool groupSync,
+                                                     bool isAllBarrier) {
+  // Execution Barrier scope:
+  // Device    = 0x1 = 1
```

Comment:
Scopes are values, not bitmasks. So here 1/2 is not converted from a hexadecimal value.

Reply:
done

Reply:
Done